src/editor/components/components/StreetSegmentSidebar.js

```diff
@@ -0,0 +1,90 @@
+import PropTypes from 'prop-types';
+import PropertyRow from './PropertyRow';
+
+const StreetSegmentSidebar = ({ entity }) => {
+  const componentName = 'street-segment';
+  // Check if entity and its components exist
+  const component = entity?.components?.[componentName];
+
+  return (
+    <div className="segment-sidebar">
+      <div className="segment-controls">
+        <div className="details">
+          {component && component.schema && component.data && (
+            <>
+              <PropertyRow
+                key="type"
+                name="type"
+                label="Segment Type"
+                schema={component.schema['type']}
+                data={component.data['type']}
+                componentname={componentName}
+                isSingle={false}
+                entity={entity}
+              />
+              <PropertyRow
+                key="width"
+                name="width"
+                label="Width"
+                schema={component.schema['width']}
+                data={component.data['width']}
+                componentname={componentName}
+                isSingle={false}
+                entity={entity}
+              />
+              <PropertyRow
+                key="direction"
+                name="direction"
+                label="Direction"
+                schema={component.schema['direction']}
+                data={component.data['direction']}
+                componentname={componentName}
+                isSingle={false}
+                entity={entity}
+              />
+              <div className="propertyRow">
+                <div className="text">-----</div>
```

Comment:
If you want some separation between set of fields, can we do something more styled? styled hr tag or just style the div with border bottom for the full width instead of this ----- text.

Reply:
yes this is a placeholder